docs/specs/config.yml

```diff
@@ -72,6 +72,19 @@ outputs:
   docs/b.json: |
     { "metadata": { "key1": "value", "key2": 2 } }
 ---
+# Global metadata schema violation
+inputs:
+  docfx.yml: |
+    globalMetadata:
+      title: 
+        - 1
+        - 2
+  docs/a.md:
+  docs/b.md:
+outputs:
+  build.log: |
+    ["error","violate-schema","Error reading string. Unexpected token: StartArray. Path 'title'.","docfx.yml"]
```

Comment:
For file metadata, since it has a nesting structure of `{metadata-name} -> {glob} -> {metadata-value}`, we can not just use Json.Net for validation, so came up with some manual validation.
For global metadata and other schema validations, the error message has been rewritten.

Reply:
Per offline discussion, line info would be enough to guide the user.